### ISSUE TITLE
Fix Request: false positive on 'unconditional creation of log argument'

### ISSUE THREAD

**Issue by rodrigolopes** [jborgers/PMD-jPinpoint-rules#176]
The following class fails validation with pmd:UnconditionalCreatedLogArguments rule.
The issue is on line `LOGGER.info("encoded value is {}", encodedValue);`
Note that `encodedValue` is not generated just for logging.

```
package com.x;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class TestController { 

    private static final Logger LOGGER = LoggerFactory.getLogger(TestController.class);
    private static final String ALGORITHM = "PBEWITHMD5ANDTRIPLEDES";

    @PostMapping("/doencoding")
    public String encode(@RequestParam("inputValue") String inputValue, @RequestParam("salt") String salt, Model model) {
        LOGGER.info("About to encode: {}", inputValue);

        String encodedValue = encodeString(inputValue, salt, ALGORITHM);

        LOGGER.info("encoded value is {}", encodedValue);

        return "encode";
    }

    public String encodeString(String value, final String salt, final String algorithm) {
        String encodedValue = ""; // do some work to encode input value
        return encodedValue;
    }
}
````

**Comment by jborgers** [comment]
The issue is that (in original version) the var used in the Log.info is also used for non-logging, via constructing a new object.
Simpler example:

```
class Klass {

    Logger log = new Logger("Klass");

    Wrapper foo() {
        String val = calcVal();
        log.info("value is=" + val);
        Wrapper wrapper = new Wrapper(val);
        return wrapper
    }

    String calcVal() {
        // calculate val work
        return calculatedVal;
    }
}
```
